app/src/userpages/components/ProfilePage/DeleteAccount/index.jsx

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import React from 'react'
+import { connect } from 'react-redux'
+import { Translate, I18n } from 'react-redux-i18n'
+import { Button } from 'reactstrap'
+
+import DeleteAccountDialog from './DeleteAccountDialog'
+import { deleteUserAccount } from '$shared/modules/user/actions'
+import { selectDeletingUserAccount } from '$shared/modules/user/selectors'
+import type { StoreState } from '$shared/flowtype/store-state'
+
+import styles from './deleteAccount.pcss'
+
+type State = {
+    modalOpen: boolean,
+}
+
+type StateProps = {
+    deletingUserAccount: boolean,
+}
+type DispatchProps = {
+    deleteAccount: () => Promise<null>,
+}
+
+type Props = StateProps & DispatchProps
+
+class DeleteAccount extends React.Component<Props, State> {
+    state = {
+        modalOpen: false,
+    }
+
+    openModal = () => {
+        this.setState({
+            modalOpen: true,
+        })
+    }
+
+    onClose = () => {
+        this.setState({
+            modalOpen: false,
+        })
+    }
+
+    onSave = () => {
+        this.props.deleteAccount()
```

Comment:
Good practice to always return promises from functions that trigger async behaviour. It's easy to miss the lack of a `return` or use of `{}` over `()`, which can make it tricky to figure out why e.g. `await this.onSave(); doSomething()` runs in the wrong order, or why `try { await this.onSave() } catch () {}` won't actually catch any async errors. Not essential, but IMO establishing a habit of good hygiene around all async behaviour costs little, and helps avoid many subtle timing issues that can be hard to debug.

Reply:
Yeah, it's good advice. In this specific case the `Dialog` and its `Buttons` component expects a `() => void` type. I changed the type to `() => void | Promise<void>` although the promise is not handled. The `Buttons` component could be actually refactored to set the spinner automatically if the `onClick` prop is a promise.